test/e2e/aws_test.go

```diff
@@ -133,13 +184,275 @@ var _ = Describe("functional tests", func() {
+			TerminationGracePeriodSeconds: &statefulsetinfo.podTerminationGracePeriod,
+			Containers: []corev1.Container{
+				{
+					Name:  statefulsetinfo.containerName,
+					Image: statefulsetinfo.containerImage,
+					Ports: []corev1.ContainerPort{{Name: statefulsetinfo.svcPortName, ContainerPort: statefulsetinfo.containerPort}},
+					VolumeMounts: []corev1.VolumeMount{
+						{Name: statefulsetinfo.volumeName, MountPath: statefulsetinfo.volMountPath},
+					},
+				},
+			},
+		},
+	}
+	return podTemplateSpec
+}
+
+func createStorageClass(storageClassName string, k8sclient crclient.Client) {
+	fmt.Fprintf(GinkgoWriter, "Creating StorageClass object with name: %s\n", storageClassName)
+	volExpansion := true
+	bindingMode := storagev1.VolumeBindingImmediate
+	storageClass := storagev1.StorageClass{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "storage.k8s.io/v1",
+			Kind:       "StorageClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: storageClassName,
+		},
+		Parameters: map[string]string{
+			"type": "gp2",
+		},
+		Provisioner:          "kubernetes.io/aws-ebs",
+		AllowVolumeExpansion: &volExpansion,
+		MountOptions:         []string{"debug"},
+		VolumeBindingMode:    &bindingMode,
+	}
+	Expect(k8sclient.Create(context.TODO(), &storageClass)).NotTo(HaveOccurred())
+}
+
+func createService(svcName string, svcNamespace string, labels map[string]string, serviceSpec corev1.ServiceSpec, k8sClient crclient.Client) {
+	svcToCreate := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: svcNamespace,
+			Name:      svcName,
+		},
+		Spec: serviceSpec,
+	}
+	if labels != nil && len(labels) > 0 {
+		svcToCreate.ObjectMeta.Labels = labels
+	}
+	Expect(k8sClient.Create(context.TODO(), &svcToCreate)).NotTo(HaveOccurred())
+}
+
+func createLBService(svcNamespace string, svcName string, k8sclient crclient.Client) string {
+	fmt.Fprintf(GinkgoWriter, "Creating service of type Load Balancer with name: %s under namespace: %s\n", svcName, svcNamespace)
+	svcSpec := corev1.ServiceSpec{
+		Type: corev1.ServiceTypeLoadBalancer,
+		Ports: []corev1.ServicePort{
+			{
+				Port:     80,
+				Protocol: corev1.ProtocolTCP,
+			},
+		},
+		Selector: map[string]string{
+			"app": "nginx",
+		},
+	}
+	createService(svcName, svcNamespace, nil, svcSpec, k8sclient)
+	//this sleep is required for the service to get updated with ingress details
+	time.Sleep(15 * time.Second)
+	svcCreated := &corev1.Service{}
+	err := k8sclient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: svcNamespace, Name: svcName}, svcCreated)
+	Expect(err).NotTo(HaveOccurred())
+	elbName := ""
+	if lbs := len(svcCreated.Status.LoadBalancer.Ingress); lbs > 0 {
+		ingressHostname := svcCreated.Status.LoadBalancer.Ingress[0].Hostname
+		elbName = strings.Split(ingressHostname, "-")[0]
+	}
+	fmt.Fprintf(GinkgoWriter, "Created Load Balancer service and ELB name is: %s\n", elbName)
+	return elbName
+}
+
+func verifyElbExists(elbName string, exists bool) {
```

Comment:
Suggestion:
It can be modified as `ifElbExist(elbName string) bool` which returns true when elb exists otherwise false

Reply:
It can be done. Then we need to do the assertion of the value returned in the "It". Even though it can be done inside "It", to follow the previously added tests convention, used this approach where assertion will happen inside the called function.